src/lxc/confile.c

```diff
@@ -2073,7 +2073,27 @@ static int set_config_rootfs_mount(const char *key, const char *value,
 static int set_config_rootfs_options(const char *key, const char *value,
 				     struct lxc_conf *lxc_conf, void *data)
 {
-	return set_config_string_item(&lxc_conf->rootfs.options, value);
+	int ret;
+	unsigned long pflags = 0;
+	struct lxc_rootfs *rootfs = &lxc_conf->rootfs;
+
+	ret = set_config_string_item(&rootfs->options, value);
+	if (ret < 0)
+		return -EINVAL;
+
+	ret = parse_mntopts(value, &rootfs->mountflags, &rootfs->data);
```

Comment:
When moving the assignment down, this should be parsed into a temporary value first. Otherwise we get mixed values.
Iow. first do all the checking, then do the assignments together.